contracts/core/EntryPoint.sol

```diff
@@ -764,60 +764,55 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuardT
-            if (paymaster == address(0)) {
-                refundAddress = mUserOp.sender;
-            } else {
+            uint256 postOpGasUsed;
+            address refundAddress = mUserOp.sender;
+            if (paymaster != address(0)) {
                 refundAddress = paymaster;
-                if (context.length > 0) {
-                    actualGasCost = actualGas * gasPrice;
-                    uint256 postOpPreGas = gasleft();
-                    if (mode != IPaymaster.PostOpMode.postOpReverted) {
-                        try IPaymaster(paymaster).postOp{
-                            gas: mUserOp.paymasterPostOpGasLimit
-                        }(mode, context, actualGasCost, gasPrice)
-                        // solhint-disable-next-line no-empty-blocks
-                        {} catch {
-                            bytes memory reason = Exec.getReturnData(REVERT_REASON_MAX_LEN);
-                            revert PostOpReverted(reason);
-                        }
-                    }
-                    // Calculating a penalty for unused postOp gas
-                    uint256 postOpGasUsed = postOpPreGas - gasleft();
-                    postOpUnusedGasPenalty = _getUnusedGasPenalty(postOpGasUsed, mUserOp.paymasterPostOpGasLimit);
+            }
+            if (paymaster != address(0) && context.length > 0 && mode != IPaymaster.PostOpMode.postOpReverted) {
+                actualGasCost = actualGas * gasPrice;
+                uint256 postOpPreGas = gasleft();
+                try IPaymaster(paymaster).postOp{
+                    gas: mUserOp.paymasterPostOpGasLimit
+                }(mode, context, actualGasCost, gasPrice)
+                // solhint-disable-next-line no-empty-blocks
+                {} catch {
+                    bytes memory reason = Exec.getReturnData(REVERT_REASON_MAX_LEN);
+                    revert PostOpReverted(reason);
                 }
+                postOpGasUsed = postOpPreGas - gasleft();
             }
+            // Calculating a penalty for unused postOp gas
+            postOpUnusedGasPenalty = _getUnusedGasPenalty(postOpGasUsed, mUserOp.paymasterPostOpGasLimit);
             actualGas += preGas - gasleft() + postOpUnusedGasPenalty;
             actualGasCost = actualGas * gasPrice;
             uint256 prefund = opInfo.prefund;
-            if (prefund < actualGasCost) {
-                if (mode == IPaymaster.PostOpMode.postOpReverted) {
-                    actualGasCost = prefund;
-                    emitPrefundTooLow(opInfo);
-                    emitUserOperationEvent(opInfo, false, actualGasCost, actualGas);
-                } else {
-                    assembly ("memory-safe") {
-                        mstore(0, INNER_REVERT_LOW_PREFUND)
-                        revert(0, 32)
-                    }
+            bool success = mode == IPaymaster.PostOpMode.opSucceeded;
+
+            if (prefund < actualGasCost && mode != IPaymaster.PostOpMode.postOpReverted) {
```

Comment:
also here: why repeating a test (prefund < actualGascost) is considered "simplify" ?